### PR TITLE
Update to 1.0.1

### PR DESCRIPTION
**Pull request by Jrice1317**
opentsne 1.0.1

**Destination channel:** {defaults}

### Links

- [PKG-4277](https://anaconda.atlassian.net/browse/PKG-4277) 
- [Upstream repository](https://github.com/pavlin-policar/openTSNE/tree/v1.0.1)

### Explanation of changes:

- Updated `version` and `sha256`
- Added `--no-build-isolation`
- Added optional dependency: `pynndescent`
- Pinned `fftw` as version >3 needed
- Added `pip`, `pip check` and `description`

[PKG-4277]: https://anaconda.atlassian.net/browse/PKG-4277?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by JamesRobertsonGames — APPROVED** [review]
Looks good to me 👍

File: recipe/meta.yaml [hunk 1]
```diff
@@ -23,27 +23,34 @@ requirements:
     - setuptools
     - wheel
     - cython
-    - fftw
     - {{ pin_compatible('numpy') }}
+    - fftw 3.3.9
   run:
+    - python
     - _openmp_mutex  # [linux]
     - llvm-openmp    # [osx]
```
Comment (SC426) [inline recipe/meta.yaml:31]: I don't think these two are needed anymore.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -23,27 +23,34 @@ requirements:
...
-    - fftw
+    - fftw >=3
     - numpy
-    - python
     - scikit-learn >=0.20
     - scipy
+    - pynndescent >=0.5.0,<0.6.0
```
Comment (SC426) [inline recipe/meta.yaml:38]: Maybe move this and `fftw` to `run_constrained`?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -23,27 +23,34 @@ requirements:
     - setuptools
     - wheel
     - cython
-    - fftw
     - {{ pin_compatible('numpy') }}
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
    - numpy {{ numpy }}
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -23,27 +23,34 @@ requirements:
...
+    - fftw 3.3.9
   run:
+    - python
     - _openmp_mutex  # [linux]
     - llvm-openmp    # [osx]
-    - fftw
+    - fftw >=3
     - numpy
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
    - {{ pin_compatible('numpy') }}
```

File: recipe/meta.yaml (hunk 2 above)
Comment (cbouss) [inline recipe/meta.yaml:38]:
```suggestion
  run_constrained:
    - hnswlib >=0.4.0,<0.5.0
    - pynndescent >=0.5.0,<0.6.0
```

File: recipe/meta.yaml (hunk 1 above)
Comment (cbouss) [reply recipe/meta.yaml:31]: setup.py does look for openmp
https://github.com/pavlin-policar/openTSNE/blob/v1.0.1/setup.py#L164

File: recipe/meta.yaml [hunk 5]
```diff
@@ -23,27 +23,34 @@ requirements:
...
     - cython
-    - fftw
     - {{ pin_compatible('numpy') }}
+    - fftw 3.3.9
   run:
+    - python
     - _openmp_mutex  # [linux]
     - llvm-openmp    # [osx]
-    - fftw
+    - fftw >=3
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
    - fftw # bound set through run_exports
```
see: https://conda-metadata-app.streamlit.app/?q=pkgs%2Fmain%2Flinux-64%2Ffftw-3.3.9-h5eee18b_2.conda

File: recipe/meta.yaml (hunk 2 above)
Comment (Jrice1317) [reply recipe/meta.yaml:38]: `fftw` is needed at runtime, but I've moved `pynndescent` to `run_constrained`. Thanks!

File: recipe/meta.yaml (hunk 1 above)
Comment (SC426) [reply recipe/meta.yaml:31]: Oops! Thanks Charles!

**Review by SC426 — APPROVED** [review]
LGTM!

**Pull request merged**